### PR TITLE
Merge flex reparent strategies

### PR DESCRIPTION
**Pull request by gbalint**
**Problem:**
absolute-reparent-to-flex and flex-reparent-to-flex strategies are very similar, they are easy to merge.
I'm working on reparent-to-flow, and I would like to start from reparent-to-flex strategies, but I don't want to separate these strategies by their original layout, because that would result in just too many very similar strategies.

**Fix:**
I merged the two strategies into a single one.

### CODE REVIEW

File: editor/src/components/canvas/canvas-strategies/strategies/reparent-to-flex-strategy.tsx [hunk 1]
```diff
@@ -35,22 +36,29 @@ export function baseAbsoluteReparentToFlexStrategy(
...
+    const isOriginallyAbsolute =
       MetadataUtils.findElementByElementPath(
         canvasState.startingMetadata,
         filteredSelectedElements[0],
-      )?.specialSizeMeasurements.position !== 'absolute'
-    ) {
+      )?.specialSizeMeasurements.position === 'absolute'
+
+    const isOriginallyFlex =
+      MetadataUtils.isParentYogaLayoutedContainerAndElementParticipatesInLayout(
```
Comment (Rheeseyb) [inline editor/src/components/canvas/canvas-strategies/strategies/reparent-to-flex-strategy.tsx:46]: We could probably prefix this with `!isOriginallyAbsolute && `

File: editor/src/components/canvas/canvas-strategies/strategies/reparent-to-flex-strategy.tsx [hunk 2]
```diff
@@ -35,22 +36,29 @@ export function baseAbsoluteReparentToFlexStrategy(
...
+        selectedElements[0],
```
Comment (Rheeseyb) [inline editor/src/components/canvas/canvas-strategies/strategies/reparent-to-flex-strategy.tsx:47]: I think this should be `filteredSelectedElements`

File: editor/src/components/canvas/canvas-strategies/strategies/reparent-to-flex-strategy.tsx [hunk 3]
```diff
@@ -75,12 +83,14 @@ export function baseAbsoluteReparentToFlexStrategy(
       ],
       fitness: fitness,
       apply: () => {
-        return applyFlexReparent(
-          'strip-absolute-props',
-          canvasState,
-          interactionSession,
-          reparentTarget,
-        )
+        return interactionSession == null
```
Comment (Rheeseyb) [inline editor/src/components/canvas/canvas-strategies/strategies/reparent-to-flex-strategy.tsx:86]: This check should be unnecessary as we've already ensured that `interactionSession` is not null earlier on in the factory

**Review by Rheeseyb — APPROVED** [review]

File: editor/src/components/canvas/canvas-strategies/strategies/reparent-to-flex-strategy.tsx (hunk 3 above)
Comment (gbalint) [reply editor/src/components/canvas/canvas-strategies/strategies/reparent-to-flex-strategy.tsx:86]: The source of mistake is that I started from the flex-reparent-to-flex strategy, and that did not use filteredSelectedElements at all

**Pull request merged**